Branch-SDK/src/main/java/io/branch/referral/BillingGooglePlay.kt

```diff
@@ -0,0 +1,273 @@
+package io.branch.referral
+
+import android.content.Context
+import com.android.billingclient.api.*
+import io.branch.indexing.BranchUniversalObject
+import io.branch.referral.util.*
+
+class BillingGooglePlay private constructor() {
+
+    lateinit var billingClient: BillingClient
+
+    companion object {
+        @Volatile
+        private lateinit var instance: BillingGooglePlay
+
+        fun getInstance(): BillingGooglePlay {
+            synchronized(this) {
+                if (!::instance.isInitialized) {
+                    instance = BillingGooglePlay()
+
+                    instance.billingClient = BillingClient.newBuilder(Branch.getInstance().applicationContext)
+                        .setListener(instance.purchasesUpdatedListener)
+                        .enablePendingPurchases()
+                        .build()
+                }
+                return instance
+            }
+        }
+    }
+
+    fun startBillingClient(callback: (Boolean) -> Unit) {
+
```

Comment:
Because this is called from different places, can you do a check to verify if you need to start (or restart) the connection? Probably use `billingClient.isReady`.